File: Loom.Core/Interfaces/IValidationResult.vb
Namespace Interfaces
    ''' <summary>
    ''' Represents the formal outcome of the invocation
    ''' </summary>

    Public Interface IValidationResult
        ''' <summary>
        ''' Indicates if the invocation is ready to be sent
        ''' </summary>
        ReadOnly Property IsValid As Boolean

        ''' <summary>
        ''' Error message list
        ''' </summary>
        ReadOnly Property Errors As IEnumerable(Of String)
    End Interface
End Namespace

File: implementations/csharp/src/Loom.Core/Interfaces/IValidationResult.cs
using System.Collections.Generic;

namespace Loom.Core.Interfaces
{
    /// <summary>
    /// Represents the formal outcome of the invocation
    /// </summary>
    public interface IValidationResult
    {
        /// <summary>
        /// Indicates if the invocation is ready to be sent
        /// </summary>
        bool IsValid { get; }

        /// <summary>
        /// Error message list
        /// </summary>
        IEnumerable<string> Errors {  get; }

    }
}